wiring/inc/spark_wiring_ble.h

```diff
@@ -406,6 +417,17 @@ class BleCharacteristic {
         return setValue(buf, len);
     }
 
+    template<typename T>
+    typename std::enable_if<std::is_integral<T>::value, ssize_t>::type
+    setValue(T val, bool ack) {
+        uint8_t buf[BLE_MAX_ATTR_VALUE_PACKET_SIZE];
+        size_t len = std::min(sizeof(T), (unsigned)BLE_MAX_ATTR_VALUE_PACKET_SIZE);
+        for (size_t i = 0, j = len - 1; i < len; i++, j--) {
+            buf[i] = reinterpret_cast<const uint8_t*>(&val)[j];
+        }
+        return setValue(buf, len, ack);
```

Comment:
What's the problem with simple `setValue(reinterpret_cast<const uint8_t*>(&val), std::min(sizeof(T), BLE_MAX_ATTR_VALUE_PACKET_SIZE), ack)` ?